common/models/role-mapping.js

```diff
@@ -42,9 +44,10 @@ module.exports = function(RoleMapping) {
    * @param {User} user
    */
   RoleMapping.prototype.user = function(callback) {
+    var RoleMapping = this.constructor;
     if (this.principalType === RoleMapping.USER) {
-      var userModel = this.constructor.User ||
-        loopback.getModelByType(loopback.User);
+      var userModel = RoleMapping.User ||
+        RoleMapping.registry.getModelByType(loopback.User);
```

Comment:
Seems that the usage of global models was not fixed/removed yet, I am not going to post any more comments about this.